src/ray/stats/stats_test.cc

```diff
@@ -88,6 +87,36 @@ TEST_F(StatsTest, F) {
   }
 }
 
+TEST_F(StatsTest, InitializationTest) {
+  // Do initialization multiple times and make sure only the first initialization
+  // was applied.
+  ASSERT_TRUE(ray::stats::StatsConfig::instance().IsInitialized());
+  auto test_tag_value_that_shouldnt_be_applied = "TEST";
+  for (size_t i = 0; i < 20; ++i) {
+    std::shared_ptr<stats::MetricExporterClient> exporter(
+        new stats::StdoutExporterClient());
+    ray::stats::Init({{stats::LanguageKey, test_tag_value_that_shouldnt_be_applied}},
+                     10054, exporter);
+  }
+
+  auto &first_tag = ray::stats::StatsConfig::instance().GetGlobalTags()[0];
+  ASSERT_TRUE(first_tag.second != test_tag_value_that_shouldnt_be_applied);
+
+  ray::stats::Shutdown();
+  ASSERT_FALSE(ray::stats::StatsConfig::instance().IsInitialized());
+
+  // Reinitialize. It should be initialized now.
+  const stats::TagsType global_tags = {
```

Comment:
We'd better add the following case for testing :
* Add mock client1 and init stats
* Shutdown stats
* Add mock client2 and init stats
* Assume only the last client2 will export data.

Reply:
Actually, I think this is unnecessary. `stats::Init` should be used only once when the process starts, and shutdown is called only once when it is terminated. The following scenario is only useful for testing.

Reply:
If you think it is important to test this for testing (which also makes sense), I will add that test. So just lmk!

Reply:
> If you believe it is important to test this scenario for testing (which also makes sense), I will add that test. So just lmk!

It's useful to protect stats module from violation.

Reply:
There's still an issue because harvesting thread is not properly cleaned up. I don't think this will affect the actual Ray (because Init shutdown should happen only once). I will add it to the backlog for now because it requires some patches on delta producer as well. I added TODO instead.

Reply:
> There's still an issue because harvesting thread is not properly cleaned up. I don't think this will affect the actual Ray (because Init shutdown should happen only once). I will add it to the backlog for now because it requires some patches on delta producer as well. I added TODO instead.

Yes. It won't affect any worker or raylet, but I'm not sure whether it makes users confused cause of thread execption might be throwed out.